### PR TITLE
specific arch for macOS pkg

### PR DESCRIPTION
**Pull request by jmartin-r7**
Thanks to @Ishaanahuja7 for [the reference](https://github.com/rapid7/metasploit-omnibus/pull/182#issuecomment-1581470436).

This change should ensure architecture specific macOS builds can be produced for x86_64 and arm64 hardware.

### CODE REVIEW

**Comment by adfoster-r7** [comment]
> Metasploit Omnibus Compilation/Packaging test — Failed to pass tests.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

File: config/software/metasploit-framework.rb [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
 # Also whitelist sqlite deps too as libz is provided just not first on path
 whitelist_file "#{install_dir}/embedded/lib/ruby/gems/#{ruby_abi_version}/gems/sqlite3-.*"
+whitelist_file "#{install_dir}//embedded/lib/ruby/gems/#{ruby_abi_version}/.*/sqlite3.*"
```
Comment (adfoster-r7) [inline config/software/metasploit-framework.rb:40]: Just confirming if this should be:

```suggestion
whitelist_file "#{install_dir}/embedded/lib/ruby/gems/#{ruby_abi_version}/.*/sqlite3.*"
```

Comment (jmartin-r7) [reply config/software/metasploit-framework.rb:40]: Yes the `//` is required, the match fails to allow the files otherwise.

Comment (adfoster-r7) [reply config/software/metasploit-framework.rb:40]: Is there any context why? 👀 

I would've assumed both of the whitelist_file call would have the same search prefix

**Review by adfoster-r7 — APPROVED** [review]

**Pull request merged**